design-documents/graph-ql/coverage/re-captcha.graphqls

```diff
@@ -0,0 +1,31 @@
+enum ReCaptchaVersion {
+  V3
+}
+
+enum ReCaptchaForms {
+  APPLY_GIFT_CARD
+  APPLY_COUPON
+  CHANGE_PASSWORD
+  CHECKOUT
+  LOGIN
+  CREATE_CUSTOMER
+  EDIT_CUSTOMER
+  NEWSLETTER
+}
+
+type ReCaptchaConfiguration {
+  type: ReCaptchaVersion!
+  website_key: String!
+  minimum_score: Float!
+  badge_position: String!
+  language_code: String
+  failure_message: String!
```

Comment:
Why is this required?  What if there's no failure?

Reply:
Or is this the failure message that's configured to display if a captcha is failed, not if something goes wrong with the config retrieval?

Reply:
Following [user guide docs](https://docs.magento.com/user-guide/configuration/security/google-recaptcha-storefront.html?itm_source=merchdocs&itm_medium=quick_search&itm_campaign=federated_search&itm_term=recaptch#recaptcha-v3-invisible) this appeared non-nullable to me.

Reply:
Shouldn't language code be required as well then?

Reply:
>Or is this the failure message that's configured to display if a captcha is failed, not if something goes wrong with the config retrieval?

Yep, exactly. If their score is under the threshold, we will prevent further submissions of the form and display this message.

>Shouldn't language code be required as well then?

That one should be safe as nullable; default behavior is to display the badge using the browser detected language, but you can optionally pass this override if you want.